content/webapp/components/ItemRequestModal/RequestDialog.tsx

```diff
@@ -82,13 +82,19 @@ const RequestDialog: FunctionComponent<RequestDialogProps> = ({
   setIsActive,
   currentHoldNumber,
 }) => {
-  const firstAvailableDate = item.availableDates
-    ? item.availableDates[0]
-    : undefined;
   const [pickUpDate, setPickUpDate] = useState<string | undefined>(
-    firstAvailableDate && dateAsValue(new Date(firstAvailableDate.from))
+    item.availableDates && dateAsValue(new Date(item.availableDates[0].from))
   );
 
+  // the RequestingDayPicker's state sometimes get set as undefined before the availableDates have been fetched
```

Comment:
Maybe we could say "`pickUpDate`'s state" instead, but I'm close to the issue so maybe this is clearer?